File: heartbeat-system/backend/app/schemas/device.py
"""
裝置相關的資料結構 (device.py)
功能: 定義裝置相關的Pydantic模型
相依: pydantic庫
"""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class DeviceBase(BaseModel):
    """裝置基本資料結構"""
    device_uid: str


class DeviceBind(DeviceBase):
    """裝置綁定請求資料結構"""
    pass


class Device(DeviceBase):
    """裝置完整資料結構"""
    id: int
    user_id: int
    
    class Config:
        orm_mode = True


class DeviceStatus(BaseModel):
    """裝置狀態資料結構"""
    id: int
    device_uid: str
    is_online: bool
    last_heartbeat: Optional[datetime] = None
    last_bpm: Optional[int] = None
    last_temperature: Optional[float] = None
    
    class Config:
        orm_mode = True

File: heartbeat-system/frontend/lib/models/device.dart
// 裝置模型 (device.dart)
// 功能: 定義裝置資料模型
// 相依: 無

class Device {
  final int id;
  final String deviceUid;
  final int userId;
  final bool isConnected;  // 添加連接狀態
  
  Device({
    required this.id,
    required this.deviceUid,
    required this.userId,
    this.isConnected = false,  // 默認為未連接
  });
  
  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: json['id'],
      deviceUid: json['device_uid'],
      userId: json['user_id'],
      isConnected: json['is_connected'] ?? false,  // 解析連接狀態
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'device_uid': deviceUid,
      'user_id': userId,
      'is_connected': isConnected,
    };
  }
}

class DeviceStatus {
  final int id;
  final String deviceUid;
  final bool isOnline;
  final DateTime? lastHeartbeat;
  final int? lastBpm;
  final double? lastTemperature;
  
  DeviceStatus({
    required this.id,
    required this.deviceUid,
    required this.isOnline,
    this.lastHeartbeat,
    this.lastBpm,
    this.lastTemperature,
  });
  
  factory DeviceStatus.fromJson(Map<String, dynamic> json) {
    return DeviceStatus(
      id: json['id'],
      deviceUid: json['device_uid'],
      isOnline: json['is_online'],
      lastHeartbeat: json['last_heartbeat'] != null ? DateTime.parse(json['last_heartbeat']) : null,
      lastBpm: json['last_bpm'],
      lastTemperature: json['last_temperature'],
    );
  }
}
